app/streaming/video/ffmpeg-renderers/d3d11va.cpp

```diff
@@ -698,14 +1239,189 @@ void D3D11VARenderer::renderVideo(AVFrame* frame)
+        m_DeviceContext->DrawIndexed(6, 0, 0);
+    }
+}
+
+/**
+ * \brief Add the Video Processor to the pipeline
+ *
+ * Creating a Video Processor add additional GPU video processing method like AI Upscaling
+ *
+ * \param bool reset default is false, at true it forces the recreate the Video Processor
+ * \return bool Returns true if the Video processor is successfully created
+ */
+bool D3D11VARenderer::createVideoProcessor(bool reset)
+{
+    HRESULT hr;
+
+    // [TODO] This timer is only used to fix a problem with VideoProcessorSetStreamColorSpace1 not properly applied at the beginning
+    // These 3 lines can be removed once the bug (non-blocking) is fixed.
+    auto now = std::chrono::system_clock::now();
+    startTime = std::chrono::duration_cast<std::chrono::milliseconds>(now.time_since_epoch()).count();
+    nextTime = startTime + increment;
+
+    D3D11_VIDEO_PROCESSOR_CONTENT_DESC content_desc;
+    ZeroMemory(&content_desc, sizeof(content_desc));
+
+    if (m_VideoProcessor.p && m_VideoProcessorEnumerator.p) {
+        hr = m_VideoProcessorEnumerator->GetVideoProcessorContentDesc(&content_desc);
+        if (FAILED(hr))
+            return false;
+
+        if (content_desc.InputWidth != m_DecoderParams.width ||
+            content_desc.InputHeight != m_DecoderParams.height ||
+            content_desc.OutputWidth != m_DisplayWidth ||
+            content_desc.OutputHeight != m_DisplayHeight || reset) {
+            m_VideoProcessorEnumerator.Release();
+            m_VideoProcessor.Release();
+        }
+        else {
+            return true;
+        }
+    }
+
+    ZeroMemory(&content_desc, sizeof(content_desc));
+    content_desc.InputFrameFormat = D3D11_VIDEO_FRAME_FORMAT_PROGRESSIVE;
+    content_desc.InputFrameRate.Numerator = m_DecoderParams.frameRate;
+    content_desc.InputFrameRate.Denominator = 1;
+    content_desc.InputWidth = m_DecoderParams.width;
+    content_desc.InputHeight = m_DecoderParams.height;
+    content_desc.OutputWidth = m_DisplayWidth;
+    content_desc.OutputHeight = m_DisplayHeight;
+    content_desc.OutputFrameRate.Numerator = m_DecoderParams.frameRate;
+    content_desc.OutputFrameRate.Denominator = 1;
+    content_desc.Usage = D3D11_VIDEO_USAGE_OPTIMAL_SPEED;
+
+    hr = m_VideoDevice->CreateVideoProcessorEnumerator(&content_desc, &m_VideoProcessorEnumerator);
+    if (FAILED(hr))
+        return false;
+
+    hr = m_VideoDevice->CreateVideoProcessor(m_VideoProcessorEnumerator, 0,
+                                             &m_VideoProcessor);
+    if (FAILED(hr))
+        return false;
+
+    m_VideoContext->VideoProcessorSetStreamAutoProcessingMode(m_VideoProcessor, 0, false);
+    m_VideoContext->VideoProcessorSetStreamOutputRate(m_VideoProcessor, 0, D3D11_VIDEO_PROCESSOR_OUTPUT_RATE_NORMAL, false, 0);
+
+    // The output surface will be read by Direct3D shaders (It seems useless in our context)
+    m_VideoContext->VideoProcessorSetOutputShaderUsage(m_VideoProcessor, true);
```

Comment:
I'm not sure what this comment means. Is it neccessary to set?

Reply:
I removed this line, it is not used.